core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala

```diff
@@ -64,6 +64,11 @@ private[deploy] object DeployMessages {
   case class WorkerSchedulerStateResponse(id: String, executors: List[ExecutorDescription],
      driverIds: Seq[String])
 
+  case class WorkerLatestState(
```

Comment:
can you add a comment here to explain when this is called, i.e. when the worker reregisters with the master.